aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -85,5 +101,6 @@ private S3Client client() {
   @Override
   public void initialize(Map<String, String> properties) {
```

Comment:
As I mention above, in the `FileIO` comment, having both this path and constructor path for initialization causes some problems.  If we use the default constructor, then `initialize()` gets called, don't we end up with two S3 client being created (and the first never gets properly closed)?